doc/src/hugo/content/guides/authors/design/patterns/command-handler-messages.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Command/Event Patterns"
+description: "Alternatives & Recommendations for Command Handler Messages"
+date: 2022-02-25T10:50:32-07:00
+draft: "false"
+weight: 10
+---
+
+## Introduction To Messages
+If you are not familiar with the concept of `Messages` as it pertains to RIDDL, see [here](../../../../../concepts/message/).
+This document pertains only to Commands and Events.
+
+## Problem Statement
+There are implementation designs to be made regarding the arrangement of input (`Command`) and output (`Event`) data 
+across [`Command Handlers`](../../../../../concepts/handler#Command%20Handler). This documents pertains all types of `Command Handlers`.
+
+These decisions can be summarized in 4 alternatives:
+1. Should separate commands on the same entity all yield the same event, containing the exact changed entity?
+2. Or should each command have its own event, containing the exact changed entity?
+3. Or should the event for each command have a version of the entity where all fields are optional, 
+such that only changed data is sent back to gateway?
+4. Or should the events for each command have only the fields that might have changed, listed individually, when replying to gateway?
+
+## Recommendations & Justifications
+
+1. This would only ensure greater obscurity in terms of being able to debug from outside - if the wrong data comes 
+back in the event, it is easier to detect a wrong event name.
+2. All commands should have unique corresponding events. They should only return the exact new version of the entity 
+on entity creation commands because all subsequent changes should have their own specific commands
+3. Versions of entities where all fields are optional should be used only for commands that are ensured to modify 
+[RIDDL Predefined Types](../../../../../language/common/types#Predefined%20Types) in the entity, except for lists. 
+This is because non-list Predefined Types are ensured to be lightweight objects.
+4. For all other commands, ie. commands that are ensured to modify lists or 
+[RIDDL Compound Types](../../../../../concepts/type#compounds), the modified data should be surfaced as individual 
+parameters of the event corresponding to the command
```

Comment:
This content is good but I would organize it differently. 

Give each of the 4 things a name and put a h3("<heading name>") in for each. Describe both the problem and solution at the same place. With four of these, and likely more to be added, separating the solution from the problem doesn't help the reader. They will want to identify the problem by the index (title of the h3), jump there, and hope to find the solution with the problem description.

Reply:
Either there is a way to generate a tree on the page based on the paragraph headers, which I can't find, or I am misunderstanding and I should be making a separate page with a separate _index.md for each point w/ example?

Reply:
YOu can generate a same-page table of contents with `{{< toc >}}` but I like your idea of putting each idea in a separate document with an example.  That way it shows up in the big index on the left and in other places.